src/matplotlib-stubs/colors.pyi

```diff
@@ -54,13 +55,21 @@ class Colormap:
     name: str
     N: int
     colorbar_extend: bool
-    def __init__(self, name: str, N:int=256) -> None: ...
+    def __init__(self, name: str, N: int = 256) -> None: ...
+    @overload
     def __call__(
         self,
-        X: float | int | np.ndarray | Scalar,
+        X: int | float,
+        alpha: float | None = None,
+        bytes: bool = False,
+    ) -> RGBAColor: ...
+    @overload
+    def __call__(
+        self,
+        X: npt.NDArray[int | float],
         alpha: float | ArrayLike | None = None,
         bytes: bool = False,
-    ) -> tuple: ...
+    ) -> npt.NDArray[RGBAColor]: ...
```

Comment:
> Here with `RGBAColor` you want to indicate the shape of the array rather that the type I think (which is not possible (not while staying reasonable at least)).

Well ... not exactly: https://numpy.org/devdocs/reference/typing.html#numpy.typing.NDArray
> Can be used during runtime for typing arrays with a given dtype and unspecified shape

--------------------

> I think it's better to have `npt.NDArray[np.uint8]` since it's the correct type, even if that means losing the shape.

It's ... more nuanced like that 🤣 

> If False (default), the returned RGBA values will be floats in the interval [0, 1] otherwise they will be [numpy.uint8](https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.uint8)s in the interval [0, 255].

We can further extend the type if you want 🤪

Reply:
I think we could have an additional overload, one with `bytes = Literal[True]` and one with `bytes = Literal[False]`. It might be a bit overkill, but if we're doing it we might as well go all the way^^"

What I meant by wanting to specify the shape rather than the type, is that `RGBAColor` is a tuple (~shape), and a tuple doesn't qualify as a dtype.

Reply:
> I think we could have an additional overload, one with `bytes = Literal[True]` and one with `bytes = Literal[False]`. It might be a bit overkill, but if we're doing it we might as well go all the way^^"

True that. And nice initiative 😃 

> What I meant by wanting to specify the shape rather than the type, is that `RGBAColor` is a tuple (~shape), and a tuple doesn't qualify as a dtype.

... I am blind with regards to numpy 😅 

However, I don't see how it is a problem. `float` is a thing, _`tuple` is another thing_. And mypy seemed to return well, so ... a _numpy array of tuples of 4 floats_? 😅

Reply:
The thing is that there's no way (afaik) to make a numpy array that contains a python tuple. The tuple will get converted to a numpy array in the process. So mypy is kinda living in a dream world.